contrib/hashicorp/vault/vault_test.go

```diff
@@ -0,0 +1,415 @@
+	}
+	spans := mt.FinishedSpans()
+	assert.Len(spans, 2)
+	span := spans[1]
+
+	// Read key error
+	assert.Equal("vault", span.Tag(ext.ServiceName))
+	assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+	assert.Equal(http.MethodGet, span.Tag(ext.HTTPMethod))
+	assert.Equal(http.MethodGet+" "+fullPath, span.Tag(ext.ResourceName))
+	assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+	assert.Equal(404, span.Tag(ext.HTTPCode))
+	assert.Equal(true, span.Tag(ext.Error))
+	assert.NotNil(span.Tag(ext.ErrorMsg))
+	assert.Nil(span.Tag("vault.namespace"))
+}
+
+func TestNamespace(t *testing.T) {
+	ts, cleanup := setupServer(t)
+	defer cleanup()
+	client, err := setupClient(ts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mountKV(client, t)()
+
+	namespace := "/some/namespace"
+	client.SetNamespace(namespace)
+	key := secretMountPath + "/testNamespace"
+	fullPath := "/v1" + key
+
+	t.Run("write", func(t *testing.T) {
+		assert := assert.New(t)
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		// Write key with namespace
+		data := map[string]interface{}{"Key1": "Val1", "Key2": "Val2"}
+		_, err = client.Logical().Write(key, data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		spans := mt.FinishedSpans()
+		assert.Len(spans, 1)
+		span := spans[0]
+
+		assert.Equal("vault", span.Tag(ext.ServiceName))
+		assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+		assert.Equal(http.MethodPut, span.Tag(ext.HTTPMethod))
+		assert.Equal(http.MethodPut+" "+fullPath, span.Tag(ext.ResourceName))
+		assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+		assert.Equal(200, span.Tag(ext.HTTPCode))
+		assert.Nil(span.Tag(ext.Error))
+		assert.Nil(span.Tag(ext.ErrorMsg))
+		assert.Equal(namespace, span.Tag("vault.namespace"))
+	})
+
+	t.Run("read", func(t *testing.T) {
+		assert := assert.New(t)
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		// Write key with namespace first
+		data := map[string]interface{}{"Key1": "Val1", "Key2": "Val2"}
+		_, err = client.Logical().Write(key, data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		// Read key with namespace
+		_, err = client.Logical().Read(key)
+		if err != nil {
+			t.Fatal(err)
+		}
+		spans := mt.FinishedSpans()
+		assert.Len(spans, 2)
+		span := spans[1]
+
+		assert.Equal("vault", span.Tag(ext.ServiceName))
+		assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+		assert.Equal(http.MethodGet, span.Tag(ext.HTTPMethod))
+		assert.Equal(http.MethodGet+" "+fullPath, span.Tag(ext.ResourceName))
+		assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+		assert.Equal(200, span.Tag(ext.HTTPCode))
+		assert.Nil(span.Tag(ext.Error))
+		assert.Nil(span.Tag(ext.ErrorMsg))
+		assert.Equal(namespace, span.Tag("vault.namespace"))
+	})
+}
+
+func TestOption(t *testing.T) {
+	ts, cleanup := setupServer(t)
+	defer cleanup()
+
+	for _, tt := range []struct {
+		name string
+		opts []Option
+		test func(assert *assert.Assertions, span mocktracer.Span)
+	}{
+		{
+			name: "Default options",
```

Comment:
This is fantastic! I think it looks really nice now!

A few notes moving forward:

* If you just have a set of sub-tests defined as `t.Run("", ...` with an empty string, the `go` tool will automatically name them `#00`, `#01`, `#02`, etc. This can be useful at times because you can refer back to the test case by index.
* If you run the tests with `go test -v` you will soon find out that test names made out of sentences (multiple words separated by space) end up looking a bit weird. For example a test named `"check default options"` will end up being displayed as `TestOption/check_default_options`, which to me is an indication that test names weren't meant to be descriptive sentences. This is further confirmed by looking at subtest names used by the Go team in the standard library. They are usually just one word, or at most two, separated by a dash, slash. In some cases they are also written as `"MultipleWords"`. 
* When naming tests, I personally feel it's a bit prettier to use a map where each key is the test name, e.g.

```go
for name, tt := range map[string][]struct { ...
```

I'm fine with either of the three modifications, can we please pick one?

Reply:
Options 2, 3 look the best. I'll do those two.